examples/ucp_client_server.c

```diff
@@ -95,6 +96,95 @@ static struct {
  */
 static void usage(void);
 
+void buffer_free(ucp_dt_iov_t *iov)
+{
+    size_t idx;
+
+    for (idx = 0; idx < iov_cnt; idx++) {
+        if (iov[idx].buffer == NULL) {
+            continue;
+        }
+        mem_type_free(iov[idx].buffer);
+        iov[idx].buffer = NULL;
+        iov[idx].length = 0;
+    }
+    mem_type_free(iov);
```

Comment:
iov itself can be simple array on the stack

Reply:
I'll check ```alloca``` API

Reply:
or just `ucp_dt_iov_t iov[NUM_IOV]`

Reply:
@yosefe should we define the ```NUM_IOV``` as constant? Currently, I use ```iov_cnt``` to store the iov counter input by user through the option ```-v```. ```ucp_dt_iov_t iov[iov_cnt];``` will repor the error ```error: ISO C90 forbids variable length array ‘iov’ [-Werror=vla]```

Reply:
ok, use alloca